components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -69,8 +69,11 @@ class SystemEngineView @JvmOverloads constructor(
      * Render the content of the given session.
      */
     override fun render(session: EngineSession) {
-        this.session = session as SystemEngineSession
+        this.session?.let {
+            removeView(it.webView)
+        }
```

Comment:
Does the method need to be synchronized?

We could also use a hammer here and remove all child views.

Reply:
Yeah, I had this with `removeAllViews` but thought it might be too much. I will change this back.

Reply:
OK, updated.